### DISCUSSION TITLE
Best way to refer from one entity span to another?

### DISCUSSION THREAD

**Discussion by kinghuang** [explosion/spaCy#10825] [Help: Coding & Implementations] [answered]
Given an entity [Span](https://spacy.io/api/span) object from [Doc.ents](https://spacy.io/api/doc#ents), what's the best way to make a reference to another entity span for describing relationships between entities? My first inclination is to just declare an extension, and assign a span to it. But, the entity span objects are ephemeral, and it feels like doing it that way isn't correct. Is there a better way to store relationships between entity spans?

**Comment by polm** [comment, marked as answer]
It's true Spans are ephemeral, but they're tied to the Doc they come from, so I don't think putting a Span under its parent Doc should be a problem. That's the strategy used in the [relation extraction component](https://github.com/explosion/projects/blob/v3/tutorials/rel_component/scripts/parse_data.py), which uses a `doc._.rel` extension, so for the relation `A eats B` you would have something like:

```
A = doc[...] # these are entity spans you got somehow
B = doc[...] 
doc._.rel = {(A, B): {"eats": 1.0}}
```

You can also use `doc.spans` for this purpose. The Spans in a SpanGroup are ordered so you can convey a relationship that way, though I think the extension might be clearer.

**Reply by kinghuang** [reply]
Thanks for the tips!
